components/org.wso2.identity.event.websubhub.publisher/src/main/java/org/wso2/identity/event/websubhub/publisher/internal/ClientManager.java

```diff
@@ -309,20 +298,29 @@ public HttpPost createHttpPost(String url, Object payload) throws WebSubAdapterE
      */
     public CompletableFuture<HttpResponse> executeAsync(HttpPost httpPost) {
 
-        //TODO: Incorporate retry mechanism
-        return CompletableFuture.supplyAsync(() -> {
-            try {
-                return getHttpAsyncClient().execute(httpPost, null).get();
-            } catch (InterruptedException ie) {
-                // Restore interrupted status
-                Thread.currentThread().interrupt();
-                throw new IdentityRuntimeException("Thread was interrupted", ie);
-            } catch (ExecutionException ee) {
-                throw new IdentityRuntimeException("Execution exception", ee);
-            } catch (Exception ex) {
-                throw new IdentityRuntimeException("Exception occurred", ex);
+        CompletableFuture<HttpResponse> future = new CompletableFuture<>();
+
+        getHttpAsyncClient().execute(httpPost, new FutureCallback<HttpResponse>() {
+            @Override
+            public void completed(HttpResponse result) {
+
+                future.complete(result);
+            }
+
+            @Override
+            public void failed(Exception ex) {
+
+                future.completeExceptionally(new IdentityRuntimeException("Execution exception", ex));
```

Comment:
Make this message more meaningful indicating in which context this asynch client gets executed

Reply:
Fix with https://github.com/wso2-extensions/identity-event-publishers/pull/38/commits/e05df994f683f6779f2f88cab919e6c137bedabe